packages/functions/src/api/service.ts

```diff
@@ -184,9 +200,15 @@ export class Service implements FirebaseFunctions {
 
     const response = await Promise.race([
       this.postJSON(url, body, headers),
-      failAfter(timeout)
+      failAfter(timeout),
+      this.cancelAllRequests
     ]);
 
+    // If service was deleted, interrupted response returns undefined.
+    if (!response) {
+      return;
```

Comment:
Returning `undefined` seems unnecessarily harsh to me. Code relying on this function returning a `Promise<HttpsCallableResult>` would crash. (It's not clear to me whether it would return `undefined` or a `Promise` that is resolved with `undefined`, actually.) But either way, if we were okay with crashing, we wouldn't be bothering to implement `delete` to begin with. So, I think it would be better to throw a proper error, as documented. Something like this:

```
      throw new HttpsErrorImpl(
        'cancelled',
        'Firebase Functions instance was deleted.'
      );
```

Reply:
Sounds good, done.